src/Compilers/CSharp/Portable/Binder/LockBinder.cs

```diff
@@ -54,6 +54,16 @@ internal override BoundStatement BindLockStatementParts(BindingDiagnosticBag dia
                 Error(diagnostics, ErrorCode.ERR_LockNeedsReference, exprSyntax, exprType);
                 hasErrors = true;
             }
+            else if (exprType.IsWellKnownTypeLock() &&
+                Compilation.GetWellKnownType(WellKnownType.System_Threading_Lock__Scope) is { } scopeType)
```

Comment:
>Compilation.GetWellKnownType(WellKnownType.System_Threading_Lock__Scope) is { } scopeType [](http://example.com/codeflow?start=16&length=89)

I think there is no guarantee that the type returned by ```Compilation.GetWellKnownType``` will have any relationship to ```exprType```. The current implementation of ```IsWellKnownTypeLock``` simply doesn't enforce that.